client/src/actions/index.js

```diff
@@ -31,7 +31,12 @@ export var submitWorkflow = (formData, history) => {
                 return "Success";
             }
         } catch (err) {
-            return history.push('/error');
+            var errorMessage = (formData.formType === "workflow" ? "An error occurred during the workflow" :
+                                                               "Failed to update profile");
+            return history.push({
+                pathname: '/error',
+                state: { message: errorMessage }
+            });
         }
```

Comment:
When a user submits a form, either from the profile or workflow, she might bump into an error. If so, we direct her to the errors page with the *appropriate* error message.

History is what manages our session's history, FYI.